modules/application/main.tf

```diff
@@ -0,0 +1,114 @@
+locals {
+  config = {
+    ca_certs = {
+      value = var.ca_certs
+    }
+
+    capacity_memory = {
+      value = var.capacity_memory
+    }
+
+    custom_image_name = {
+      value = var.custom_image_name
+    }
+
+    enc_password = {
+      value = var.enc_password
+    }
+
+    extern_vault_addr = {
+      value = var.extern_vault_addr
+    }
+
+    extern_vault_enable = {
+      value = var.extern_vault_addr == null ? "0" : "1"
+    }
+
+    extern_vault_path = {
+      value = var.extern_vault_path
+    }
+
+    extern_vault_role_id = {
+      value = var.extern_vault_role_id
+    }
+
+    extern_vault_secret_id = {
+      value = var.extern_vault_secret_id
+    }
+
+    extern_vault_token_renew = {
+      value = var.extern_vault_token_renew
+    }
+
+    extra_no_proxy = {
+      value = join(",", var.additional_no_proxy)
+    }
+
+    gcs_bucket = {
+      value = var.storage_bucket_name
+    }
+
+    gcs_credentials = {
+      # This is required because when we use the value later, it needs to be one line otherwise
+      # the templating breaks inside kubernetes. That should be addressed as well, but until then
+      # we do it here.
+      value = replace(base64decode(var.service_account_storage_key_private_key), "\n", "")
```

Comment:
could we use custom validation rules for the service_account_storage_key_private_key definition?

https://www.terraform.io/docs/configuration/variables.html#custom-validation-rules